components/static_redirect_helper/static_redirect_helper.cc

```diff
@@ -118,6 +120,13 @@ void StaticRedirectHelper(const GURL& request_url, GURL* new_url) {
     return;
   }
 
+  if (favicon_pattern->MatchesURL(request_url)) {
+    replacements.SetSchemeStr("https");
```

Comment:
we're only matching on https so do we need this?

Reply:
Not needed! 😄 
Removed in https://github.com/brave/brave-core/pull/30734/commits/5bd1b447b6a8b3b3b8cb603e1e2f49ccc35c54e6